src/main/resources/data/computercraft/lua/rom/modules/main/cc/shell/completion.lua

```diff
@@ -74,12 +76,41 @@ end
 
 --- Complete the name of a program.
 --
--- @tparam table shell The shell we're completing in
--- @tparam { string... } choices The list of choices to complete from.
+-- @tparam table shell The shell we're completing in.
+-- @tparam string text Current text to complete.
+-- @tparam { string... } previous The shell arguments before this one.
+-- @tparam[opt] boolean completion_space Whether to add a space after the completed item if there is completion for this program.
+-- @treturn { string... } A list of suffixes of matching programs.
+-- @see shell.completeProgram
+local function program(shell, text, previous, completion_space)
+    local results = shell.completeProgram(text)
+    if completion_space then
+        local tCompletionInfo = shell.getCompletionInfo()
+        for n = 1, #results do
+            local sResult = results[n]
+            if sResult:sub(-1) ~= "/" and tCompletionInfo[shell.resolveProgram(text .. sResult)] then
+                results[n] = sResult .. " "
+            end
+        end
+    end
+    return results
+end
+
+--- Complete arguments of a program.
+--
+-- @tparam table shell The shell we're completing in.
+-- @tparam string text Current text to complete.
+-- @tparam { string... } previous The shell arguments before this one.
+-- @tparam string|number program Path of program to complete args for, or a number of argument index which specified program.
+-- @tparam number starting Which argument index this program args started at.
 -- @treturn { string... } A list of suffixes of matching programs.
 -- @see shell.completeProgram
-local function program(shell, text)
-    return shell.completeProgram(text)
+local function programArgs(shell, text, previous, program, starting)
```

Comment:
I think it might be nicer here if we had `programWithArgs`, which only accepts a "starting" parameter. It's unlikely you're ever going to have a point where you have programArgs which _isn't_ preceded with a program.

Reply:
Are you thinking `programWithArgs` that completes program initially then args on next argument ones and replaces both calls or keep current `complete.program` then `complete.programWithArgs` that is just provided with starting point pointing at `complete.program` to read the program name from?

Reply:
It is certainly doable and i can make it work that way no problem.
It would just break current theme of having each function do only one thing. (completes programs, completes directories, files, both etc). 
It would be a case of function that completes a program on first call and then arguments on next `many` calls. If that is way you think it should be working its no more awkward as current monitor completion.

Reply:
> It would just break current theme of having each function do only one thing. 

Hrrm. I rather view "program and arguments" as a single unit rather than two. Now, I realise that sounds absurd, but conversely "arguments" on its own doesn't make any sense! 

Obviously this whole API is made worse by `build`'s weird quirks (or even, build's `many` quirks). In worked _fine_ at the time, but is proving increasingly inflexible